acinclude/ldap.m4

```diff
@@ -0,0 +1,98 @@
+# Copyright (C) 1996-2022 The Squid Software Foundation and contributors
+##
+## Squid software is distributed under GPLv2+ license and includes
+## contributions from numerous individuals and organizations.
+## Please see the COPYING and CONTRIBUTORS files for details.
+##
+
+dnl checks for LDAP functionality
+AC_DEFUN([SQUID_LDAP_TEST],[
+  AC_CACHE_CHECK([for $1],[squid_cv_$1],[
+    SQUID_STATE_SAVE(squid_ldap_test_state)
+    LIBS="$LIBLDAP_PATH $LIBLDAP_LIBS $LIBPTHREADS"
+    CXXFLAGS="-DLDAP_DEPRECATED=1 -DLDAP_REFERRALS $CXXFLAGS"
+    AC_COMPILE_IFELSE([AC_LANG_PROGRAM([[
+#       if HAVE_LDAP_H
+#       include <ldap.h>
+#       elif HAVE_MOZLDAP_LDAP_H
+#       include <mozldap/ldap.h>
+#       endif
+      ]],[[$2]])
+      SQUID_STATE_ROLLBACK(squid_ldap_test_state)
+    ],[
+      squid_cv_$1=1
+    ],[
+      squid_cv_$1=0
+    ],[
+      squid_cv_$1=0
+    ])
+  ])
+  AC_DEFINE([HAVE_$1],[${squid_cv_$1}],[Define to 1 if you have $1])
+])
+
+dnl similar to SQUID_LDAP_TEST but runs the test program
+AC_DEFUN([SQUID_LDAP_TEST_RUN],[
+  AC_CACHE_CHECK([for $1],[m4_translit([squid_cv_$1],[-+. ],[____])],[
+    SQUID_STATE_SAVE(squid_ldap_test_state)
+    LIBS="$LIBLDAP_PATH $LIBLDAP_LIBS $LIBPTHREADS"
```

Comment:
PKG_CHECK_MODULES() sets LIBLDAP_LIBS. If the `_PATH` variant was added by accident:

```suggestion
    LIBS="$LIBLDAP_LIBS $LIBPTHREADS"
```

Reply:
Intentional change. The _PATH variable is custom set by SQUID_AUTO_LIB when the user specifies a custom --with-foo=path location for the library.